source/extensions/common/wasm/context.cc

```diff
@@ -153,12 +153,23 @@ WasmResult Buffer::copyFrom(size_t start, size_t length, std::string_view data)
 }
 
 Context::Context() = default;
-Context::Context(Wasm* wasm) : ContextBase(wasm) {}
+Context::Context(Wasm* wasm) : ContextBase(wasm) {
+  if (wasm != nullptr) {
+    abi_version_ = wasm->abi_version_;
+  }
```

Comment:
Yeah, you are completely right at the production code view. But this is helpful for our tests now to allow us to mock different abi versions.

Our tests also need to be improved to avoid this, but, I want to put this to a refactor only PR, for our tests and some Plugin context.

(Much things need to do... :(

Reply:
(optional) WDYT about defining a virtual Context::getAbiVersion() method whose default impl calls wasm()->abiVersion(), but can be overridden by TestContext?

The minor quibble I have with adding an abi_version_ field here is that now the information is stored in multiple places, as opposed to one authoritative source.

Reply:
> (optional) WDYT about instead defining a virtual Context::getAbiVersion() method whose default impl calls wasm()->abiVersion(), but can be overridden by TestContext?
> 
> The minor quibble I have with adding an abi_version_ field here is that now the information is stored in multiple places, as opposed to one authoritative source.

SGTM. But I want to did some refactoring to these code anyway, will defer this to future PR.